robot_modeling/DQ_SerialManipulator.m

```diff
@@ -67,76 +58,83 @@
         handle
         n_links;
     end
-    
+
+    methods (Abstract)
+         % RAW_POSE_JACOBIAN(q) returns the Jacobian that satisfies 
+         % vec(x_dot) = J * q_dot, where x = fkm(q) and q is the 
+         % vector of joint variables.
+         %
+         % RAW_POSE_JACOBIAN(q,ith) returns the Jacobian that
+         % satisfies vec(x_ith_dot) = J * q_dot(1:ith), where 
+         % x_ith = fkm(q, ith), that is, the fkm up to the i-th link.
+         %
+         % This function does not take into account any base or
+         % end-effector displacements and should be used mostly
+         % internally in DQ_kinematics
+         J = raw_pose_jacobian(obj, q,to_ith_link);
+
+
+         % RAW_POSE_JACOBIAN_DERIVATIVE(q,q_dot) returns the Jacobian 
+         % time derivative.
+         % 
+         % RAW_POSE_JACOBIAN_DERIVATIVE(q,q_dot, to_ith_link) returns the first
+         % to_ith_link columns of the Jacobian time derivative.
+         % This function does not take into account any base or
+         % end-effector displacements.
+         J_dot = raw_pose_jacobian_derivative(obj, q, q_dot, to_ith_link);
+       
+         %   RAW_FKM(q) calculates the forward kinematic model and
+         %   returns the dual quaternion corresponding to the
+         %   last joint (the displacements due to the base and the effector
+         %   are not taken into account).
+         %
+         %   'q' is the vector of joint variables
+         %   'to_ith_link' defines until which link the raw_fkm will be
+         %   calculated.
+         %
+         %   This is an auxiliary function to be used mainly with the
+         %   Jacobian function.   
+         pose = raw_fkm(obj,q, to_ith_link); 
+    end
+
+    methods (Abstract, Access = protected)   
+        %   DQ2DQ(q, ith) calculates  the corresponding dual quaternion for
+        %   a given link's DH parameters
+        %
+        %   Usage: dq = dh2dq(q,ith), where
+        %          q: joint value
+        %          ith: link number
+        %
+        %   Eq. (2.34) of Adorno, B. V. (2011). Two-arm Manipulation: From Manipulators
+        %   to Enhanced Human-Robot Collaboration [Contribution à la manipulation à deux bras : 
+        %   des manipulateurs à la collaboration homme-robot]. 
+        %   https://tel.archives-ouvertes.fr/tel-00641678/
+        dq = dh2dq(obj,q,ith);
```

Comment:
Hi, @juanjqo,

I realized that I gave you a bad suggestion. :cry: The `dh2dq` is only relevant to the `DH`  and `MDH` subclasses because the method is named DH to DQ, which clearly shows its relationship with the DH convention.

Therefore, may you remove the abstract signature here?

Thanks,
Bruno

Reply:
@bvadorno 
Sure! 
I did the modification in [
116bbe5](https://github.com/dqrobotics/matlab/pull/75/commits/116bbe5e3ee5f1e46c489eb9f95c05df67df1841).